src/test/txpackage_tests.cpp

```diff
@@ -413,6 +413,17 @@ BOOST_FIXTURE_TEST_CASE(package_witness_swap_tests, TestChain100Setup)
 
         BOOST_CHECK(m_node.mempool->exists(GenTxid::Txid(ptx_child2->GetHash())));
         BOOST_CHECK(!m_node.mempool->exists(GenTxid::Wtxid(ptx_child2->GetWitnessHash())));
+
+        // Deduplication should work when wtxid != txid. Submit package with the already-in-mempool
```

Comment:
>when wtxid != txid

(nonblocking:) This confused me; can this be stated more clearly? `wtxid` can never equal `txid`.
```suggestion
        // Deduplication should work when existing wtxid != package wtxid. Submit package with the already-in-mempool
```

Reply:
The `wtxid` of a transaction is equal to its `txid` when there is no witness data. This is precisely what this commit fixes and why this test didn't catch the bug earlier: `ptx_parent` is submitted multiple times, but its `txid` is equal to its `wtxid`. Otherwise it would have failed the assertion.